lib/std/http/Server.zig

```diff
@@ -352,28 +415,43 @@ pub const Response = struct {
             .zstd => |*zstd| zstd.deinit(),
         }
 
-        if (!res.request.parser.done) {
-            // If the response wasn't fully read, then we need to close the connection.
-            res.connection.conn.closing = true;
-        }
+        res.state = .start;
+        res.version = .@"HTTP/1.1";
+        res.status = .ok;
+        res.reason = null;
 
-        if (res.connection.conn.closing) {
-            res.connection.close();
+        res.transfer_encoding = .none;
 
-            if (res.request.parser.header_bytes_owned) {
-                res.request.parser.header_bytes.deinit(res.server.allocator);
-            }
+        res.headers.clearRetainingCapacity();
+
+        res.request.headers.clearRetainingCapacity();
+        res.request.parser.reset();
+
+        res.request = Request{
```

Comment:
ah ok - this will reset the request, using default values ... as you were saying above.
sry, didn't spot that one.
That makes sense.